### PR TITLE
feat(events): add route to context

### PR DESCRIPTION
**Pull request by pavelsmejkal**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it

- feat: request route to be able to describe HTTP path without extrapolated path variables. This will be used in common service heath dashboard. 
- feat: apiv2 client will be updated to leverage this feature 

## Jira ID

[XX-XX]

### CODE REVIEW

File: pkg/events/events.go [hunk 1]
```diff
@@ -81,6 +82,9 @@ type HTTPRequest struct {
 	Path       string `log:"http.url_details.path"`
 	URI        string `log:"http.url_details.uri"`
 	Endpoint   string `log:"http.url_details.endpoint"`
+
+	// Route is path without extrapolated path variables
```
Comment (malept) [inline pkg/events/events.go:86]: Nit:
```suggestion
	// Route is the URL path without interpolating the path variables.
```

File: pkg/events/events.go [hunk 2]
```diff
@@ -125,3 +130,20 @@ func (h *HTTPRequest) getRemoteAddr(r *http.Request) string {
 	// See: ETC-182.  Use a remote address library
 	return strings.Split(r.Header.Get("X-Forwarded-For"), ",")[0]
 }
+
+// requestRouteCtxKey represents the context key for the request route
```
Comment (malept) [inline pkg/events/events.go:134]: Nit:
```suggestion
// requestRouteCtxKey represents the context key for the request route.
```

File: pkg/events/events.go [hunk 3]
```diff
@@ -125,3 +130,20 @@ func (h *HTTPRequest) getRemoteAddr(r *http.Request) string {
...
+type requestRouteCtxKey struct{}
+
+// WithRequestTime adds a given time to a context as a request time
```
Comment (malept) [inline pkg/events/events.go:137]: issue: this was not changed from the copy/paste.

File: pkg/events/events.go [hunk 4]
```diff
@@ -125,3 +130,20 @@ func (h *HTTPRequest) getRemoteAddr(r *http.Request) string {
...
+func WithRequestRoute(ctx context.Context, route string) context.Context {
+	return context.WithValue(ctx, requestRouteCtxKey{}, route)
+}
+
+// RequestRoute returns a request route present in the context or empty string
```
Comment (malept) [inline pkg/events/events.go:142]: Nit:
```suggestion
// RequestRoute retrieves the request route present in the context, if available.
```

File: pkg/events/marshalers.go [hunk 5]
```diff
@@ -30,6 +30,9 @@ func (s *HTTPRequest) MarshalLog(addField func(key string, value interface{})) {
 	addField("http.url_details.path", s.Path)
 	addField("http.url_details.uri", s.URI)
 	addField("http.url_details.endpoint", s.Endpoint)
+    if s.Route != "" {
+        addField("http.route", s.Route)
+    }
```
Comment (malept) [inline pkg/events/marshalers.go:35]: issue: the indentation is inconsistent from the context.

**Review by malept — CHANGES_REQUESTED** [review]
Conceptually this seems fine, but it needs some polish.

**Review by malept — APPROVED** [review]

**Review by john-dostal — APPROVED** [review]

**Review by milankonir — APPROVED** [review]

File: pkg/events/marshalers.go [hunk 6]
```diff
@@ -30,6 +30,9 @@ func (s *HTTPRequest) MarshalLog(addField func(key string, value interface{})) {
 	addField("http.url_details.path", s.Path)
 	addField("http.url_details.uri", s.URI)
 	addField("http.url_details.endpoint", s.Endpoint)
+	if s.Route != "" {
+		addField("http.route", s.Route)
+	}
```
Comment (malept) [inline pkg/events/marshalers.go:35]: issue: as it turns out, this change is in an autogenerated file, so this change gets reverted every time `make gogenerate` is run, which also happens every time a re-stencil happens.

**Pull request merged**